### PR TITLE
front: fix pointer behaviour on charts with tsv2

### LINKED ISSUE #7577: TSv2: Pointer on the speed space chart and space curves slopes chart and horizontal synchro are broken
### What happened?

With train schedule v2 switch, when hovering on the speed space chart and on the space curves slopes chart directly (this doesn't happen in the space time chart), the pointer moves everywhere and the synchro is lost.

Also when hovering on the space time chart, the horizontal axis is not synchronized with the one on the speed space chart.

### What did you expect to happen?

Pointer should work on all charts

### How can we reproduce it (as minimally and precisely as possible)?

1. Go to a scenario
2. Move the mouse on the speed space chart or on the space curves slopes chart

### What operating system, browser and environment are you using?

- **Browser**: Firefox v126
- **OS**: macOS Sonoma 14.5
- **Env**: Dev

### OSRD version (top right corner `Account` button > `Informations`)

dev 4ae7056

### PR DESCRIPTION
**Pull request by clarani**
closes #7577

### CODE REVIEW

File: front/src/modules/simulationResult/components/ChartHelpers/ChartHelpers.ts [hunk 1]
```diff
@@ -295,6 +296,28 @@ export const interpolateOnPosition = (
...
+export const interpolateOnPositionV2 = (
+  dataSimulation: { speed: PositionSpeedTime[] | ReportTrainData[] }, // TODO DROP V1 : remove PositionSpeedTime type
+  positionLocal: number,
+  offset: number // in seconds
+) => {
+  const bisect = d3.bisector<PositionSpeedTime, number>((d) => d.position).left;
+  const index = bisect(dataSimulation.speed, positionLocal, 1);
+  const bisection = [dataSimulation.speed[index - 1], dataSimulation.speed[index]];
+  if (bisection[1]) {
+    const distanceBewteenSteps = bisection[1].position - bisection[0].position;
```
Comment (Yohh) [inline front/src/modules/simulationResult/components/ChartHelpers/ChartHelpers.ts]:
```suggestion
    const distanceBetweenSteps = bisection[1].position - bisection[0].position;
```

**Review by Yohh — CHANGES_REQUESTED** [review]
nice, the GEV V1 looks so much better!

one comment for first

**Review by SharglutDev — APPROVED** [review]
Lgtm and tested, great job !

File: front/src/modules/simulationResult/components/ChartHelpers/ChartHelpers.ts (hunk 1 above)
Comment (clarani) [reply front/src/modules/simulationResult/components/ChartHelpers/ChartHelpers.ts]: Nice spot ! Fixed ;)

**Review by Yohh — APPROVED** [review]
LGTM :muscle: tested

**Pull request merged**